### PR TITLE
Adding comment for cmdline_contains()

### PR DESCRIPTION
**Pull request by andreitraistaru**
Signed-off-by: Traistaru Andrei Cristian <atc@amazon.com>

## Reason

Adding a comment according to https://github.com/firecracker-microvm/firecracker/pull/3264#discussion_r1023828793

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] New `unsafe` code is documented.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md

### CODE REVIEW

**Review by roypat — APPROVED** [review]
(yes I realized I am complaining about my own wording here, haha. My comment was written in a rush, sorry)

File: src/vmm/src/builder.rs [hunk 1]
```diff
@@ -1089,6 +1089,16 @@ pub mod tests {
...
+        // The following unwraps can never fail; the only way any of these methods
+        // would return an Err is if one of the following conditions is met:
+        //    1. the command line is empty (we just added things to it, and if insertion
+        //       of an argument goes wrong, then CmdLine::insert* would have already
+        //       returned Err)
+        //    2. there's a spurious nul character somewhere in the command line (the
+        //       CmdLine::insert* methods verify that this is not the case)
+        //    3. the CString is not valid UTF8 (it just got created from a String, which
+        //       was valid UTF8)
+
```
Comment (JonathanWoollett-Light) [inline src/vmm/src/builder.rs:1101]:
```suggestion
        // The following unwraps can never fail; the only reason any of these methods
        // would return an `Err` is if one of the following conditions is met:
        //    1. The command line is empty: We just added things to it, and if insertion
        //       of an argument goes wrong, then `CmdLine::insert` would have already
        //       returned `Err`.
        //    2. There's a spurious null character somewhere in the command line: The
        //       `CmdLine::insert` methods verify that this is not the case.
        //    3. The `CString` is not valid UTF8: It just got created from a `String`, which
        //       was valid UTF8.
```

**Review by roypat — DISMISSED** [review]

**Review by roypat — APPROVED** [review]

File: src/vmm/src/builder.rs [hunk 2]
```diff
@@ -1089,6 +1089,15 @@ pub mod tests {
...
     fn cmdline_contains(cmdline: &Cmdline, slug: &str) -> bool {
+        // The following unwraps can never fail; the only reasons any of the methods
+        // would return an Err is if one of the following conditions are met:
+        //    1. the command line is empty (we just added things to it, and if insertion of
+        //       an argument goes wrong, then insert would have already returned Err)
+        //    2. there's a spurious nul character somewhere in the command line (the insert
+        //       methods verify that this is not the case)
+        //    3. the CString is not valid UTF8 (it just got created from a String, which
+        //       was valid UTF8)
+
```
[the remark this answers is not in the document]
Comment (andreitraistaru) [reply src/vmm/src/builder.rs]: Did the change, but I used `Cmdline::insert` instead.

File: src/vmm/src/builder.rs (hunk 1 above)
Comment (andreitraistaru) [reply src/vmm/src/builder.rs]: Did the change, with one single difference instead of CmdLine, I used Cmdline.

**Review by JonathanWoollett-Light — APPROVED** [review]

**Pull request merged**